### PR TITLE
Only attempt to get Google ad_id if using FCM

### PR DESCRIPTION
**Pull request by jkasten2**
* This cleans up the "Error getting Google Ad id"  on HMS only devices

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-android-sdk/1061)

### CODE REVIEW

**Comment by mikechoch** [comment]
I see we check the `ad_id` is `null` for the Huawei tests, should we add a UnitTest for Amazon devices checking against the `ad_id` as well? 

Maybe a separate PR, but otherwise LGTM!

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java [hunk 1]
```diff
@@ -452,7 +452,15 @@ public void onSessionEnding(@NonNull List<OSInfluence> lastInfluences) {
    @Nullable private static OSOutcomeEventsController outcomeEventsController;
    @Nullable private static OSOutcomeEventsFactory outcomeEventsFactory;
 
-   private static AdvertisingIdentifierProvider mainAdIdProvider = new AdvertisingIdProviderGPS();
+   @Nullable private static AdvertisingIdentifierProvider adIdProvider;
+   private static synchronized @Nullable AdvertisingIdentifierProvider getAdIdProvider() {
+      if (adIdProvider == null) {
+         if (OSUtils.isAndroidDeviceType())
```
Comment (Jeasmine) [inline OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:458]: maybe put on the same if?

Comment (jkasten2) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:458]: I set it up nested like this as we may want to add an `else if (OSUtils.isHuaweiDeviceType()` ect if we can support other ad providers in the future.

Comment (Jeasmine) [reply OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:458]: sounds good!

**Comment by jkasten2** [comment]
> I see we check the `ad_id` is `null` for the Huawei tests, should we add a UnitTest for Amazon devices checking against the `ad_id` as well?
> 
> Maybe a separate PR, but otherwise LGTM!

I think it would be good to have these tesst in this PR with the code changes. I'll work on that

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java [hunk 2]
```diff
@@ -322,6 +323,59 @@ public void testDeviceTypeIsAmazon_forPlayerCreate() throws Exception {
...
+      threadAndTaskWait();
+
+      // 3. Make sure device_type is Huawei (13) in player create
+      assertHuaweiPlayerCreateAtIndex(1);
+
+      // 4. Assert Player Create does NOT have an ad_id
+      ShadowOneSignalRestClient.Request request = ShadowOneSignalRestClient.requests.get(1);
+      JsonAsserts.doesNotContainKeys(request.payload, new ArrayList<String>() {{ add("ad_id"); }});
+
+      // 4. Assert we did NOT try to get a Google Ad id
```
Comment (mikechoch) [inline OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:375]: `4.` should be `5.`

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java [hunk 3]
```diff
@@ -322,6 +323,59 @@ public void testDeviceTypeIsAmazon_forPlayerCreate() throws Exception {
...
+      threadAndTaskWait();
+
+      // 3. Make sure device_type is Amazon (2) in player create
+      assertAmazonPlayerCreateAtIndex(1);
+
+      // 4. Assert Player Create does NOT have an ad_id
+      ShadowOneSignalRestClient.Request request = ShadowOneSignalRestClient.requests.get(1);
+      JsonAsserts.doesNotContainKeys(request.payload, new ArrayList<String>() {{ add("ad_id"); }});
+
+      // 4. Assert we did NOT try to get a Google Ad id
```
Comment (mikechoch) [inline OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:342]: `4.` should be `5.`

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java (hunk 2 above)
Comment (jkasten2) [reply OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java]: Done

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java (hunk 3 above)
Comment (jkasten2) [reply OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java]: Done

**Pull request merged**